packages/react-core/src/demos/ComposableMenu/examples/ComposableDropdwnVariants.tsx

```diff
@@ -0,0 +1,235 @@
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+  const [toggleSelected, setToggleSelected] = React.useState<string>('basic');
+  const menuRef = React.useRef<HTMLDivElement>();
+  const toggleRef = React.useRef<HTMLButtonElement>();
+
+  const handleToggleSwitch = (selected: boolean, e: React.MouseEvent<any> | React.KeyboardEvent | MouseEvent) => {
+    setToggleSelected(e.currentTarget.id);
+  };
+
+  const handleMenuKeys = (event: KeyboardEvent) => {
+    if (isOpen && menuRef && menuRef.current && menuRef.current.contains(event.target as Node)) {
+      if (event.key === 'Escape' || event.key === 'Tab') {
+        setIsOpen(!isOpen);
+        toggleRef.current.focus();
+      }
+    }
+  };
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (isOpen && menuRef && menuRef.current && !menuRef.current.contains(event.target as Node)) {
+      setIsOpen(false);
+    }
+  };
+
+  React.useEffect(() => {
+    window.addEventListener('keydown', handleMenuKeys);
+    window.addEventListener('click', handleClickOutside);
+
+    return () => {
+      window.removeEventListener('keydown', handleMenuKeys);
+      window.removeEventListener('click', handleClickOutside);
+    };
+  }, [isOpen, menuRef]);
+
+  const onToggleClick = (ev: React.MouseEvent) => {
+    ev.stopPropagation(); // Stop handleClickOutside from handling
+    setTimeout(() => {
+      if (menuRef.current) {
+        const firstElement = menuRef.current.querySelector('li > button,input:not(:disabled),a');
+        firstElement && (firstElement as HTMLElement).focus();
+      }
+    }, 0);
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = (ev: React.MouseEvent<Element, MouseEvent>, itemId: string) => {
+    // eslint-disable-next-line no-console
+    console.log(`Menu item ${itemId} selected`);
+    setIsOpen(false);
+  };
+
+  const buildToggle = () => {
+    switch (toggleSelected) {
+      case 'basic':
+        return (
+          <MenuToggle ref={toggleRef} onClick={onToggleClick} isExpanded={isOpen}>
+            Dropdown
+          </MenuToggle>
+        );
+      case 'kebab':
+        return (
+          <MenuToggle
+            ref={toggleRef}
+            onClick={onToggleClick}
+            isExpanded={isOpen}
+            variant="plain"
+            aria-label="plain kebab"
+          >
+            <EllipsisVIcon />
+          </MenuToggle>
+        );
+      case 'badge':
+        return (
+          <MenuToggle
+            ref={toggleRef}
+            onClick={onToggleClick}
+            isExpanded={isOpen}
+            variant="plain"
+            aria-label="plain badge"
+          >
+            <Badge>
+              4
+              <span>
+                <CaretDownIcon />
+              </span>
+            </Badge>
+          </MenuToggle>
+        );
+      case 'image':
+        return (
+          <MenuToggle
+            ref={toggleRef}
+            icon={<Avatar src={avatarImg} alt="avatar" />}
```

Comment:
It looks like we use `.pf-c-menu-toggle__image` in core for this versus `.pf-c-menu-toggle__icon`, though I don't know that we need both classes - curious if we could just use a single class for both.

Some other thoughts:
1. `.pf-c-menu-toggle__image` shouldn't be used right now since there is a bug in the spacing - the right margin for the image isn't being applied, so the image will touch the adjacent text.
2. the [dropdown image toggle image element](https://www.patternfly.org/v4/components/dropdown/html#dropdown-with-image-and-text) has some additional CSS. It applies a top/bottom margin, too, and uses `inline-flex` and `flex-shrink: 0;`. I'm not sure about the top/bottom margin, but I think the flex properties are good adds.
  * <img width="314" alt="Screen Shot 2022-01-20 at 3 02 18 PM" src="https://user-images.githubusercontent.com/35148959/150422544-97b343ac-cd9c-4061-8982-582aa1a69cb9.png">
3. the image isn't really aligned. we could apply `line-height: 1; align-self: center` to the `__image/icon` element. This is what it looks like in the PR.
  * <img width="273" alt="Screen Shot 2022-01-20 at 3 10 12 PM" src="https://user-images.githubusercontent.com/35148959/150422646-ad614bbf-ad44-4021-b7b0-6d7cf537168a.png">
 
I think we could just apply all of this to a single element that holds an icon or image. wdyt @mattnolting?

Reply:
@mcoker I actually added the `flex-shrink` prop in [4617](https://github.com/patternfly/patternfly/commit/9fec1a01f3a7edc5c284698bb9f694a6d82c8bdc#diff-27c8e819fb9e49641ffc5897a7b3e04c569a4fd870c88d937a66f0d3247ffafeR532) as the image was distorted when width became constrained. I agree, we only need one container element and it should be `__image` IMO. 

>  I'm not sure about the top/bottom margin, but I think the flex properties are good adds.

We should nix top/bottom margin

> the image isn't really aligned. we could apply line-height: 1; align-self: center to the __image/icon element. This is what it looks like in the PR.

👍 👍 👍